waspc/data/Generator/templates/server/src/routes/auth/signup.js

```diff
@@ -1,11 +1,30 @@
 {{={= =}=}}
+import Prisma from '@prisma/client'
 import prisma from '../../dbClient.js'
 import { handleRejection } from '../../utils.js'
+import AuthError from '../../core/AuthError.js'
+import HttpError from '../../core/HttpError.js'
 
 export default handleRejection(async (req, res) => {
   const userFields = req.body || {}
 
-  await prisma.{= userEntityLower =}.create({ data: userFields })
+  try {
+    await prisma.{= userEntityLower =}.create({ data: userFields })
+  } catch(e) {
+    if (e instanceof AuthError) {
+      throw new HttpError(422, 'Validation failed', { message: e.message })
+    } else if (e instanceof Prisma.PrismaClientKnownRequestError && e.code === 'P2002') {
+      // Warn about unique constraint for email already existing
+      throw new HttpError(422, 'Save failed',
+        {
+          message: `A user with the same ${e.meta.target.join(', ')} already exists.`,
+          target: e.meta.target
+        })
+    } else {
+      // Do not add any extra Prisma error info for security sake
+      throw new HttpError(422, 'Save failed')
```

Comment:
Tricky thing is that we can't be sure if this is 422 or something else -> maybe save failed because our connection to the database got lost?
Hm so what should we return -> maybe 500, since there is a possibility it is an error on our side?
It can be anything, we just don't know. Does this make sense, I am not sure if I am missing something, please let me know if so!

Reply:
This was the one that was thorny (for me) for a while. There are like 2 or 3 other possible Prisma error types, but you are right, it may not be _truly_ Prisma related as the catch all. What I will do is update it to explicitly check for those other types and do 422 only when we catch them and have the default 500 as the base. Was leaning that way, went against it for brevity, but yeah, in hindsight should not have. Thnx

Reply:
Actually, after sleep (see naps are good lol) I think we can get it pretty accurate (question is now vs later). `PrismaClientValidationError` should return `422` as well. My only final hangup is many of the `P20xx` codes would be a mix of either `422` or `500`. For example, `P2000 ` value too long for column, vs `P2021 ` table does not exist. All `P1xxx` and `P3xxx` appear to be `500`.

Thinking out loud, I wonder if I should create a helper that can determine if an error is a Prisma one `isPrismaError(e) => bool` and if so `prismaErrorToHttpError(e1) -> e2`. Side note, how does Prisma not have something similar for us already?!? I can put both of those helpers in `dbClient.js` or `utils.js` for other future controller usage. I'm sure I won't be the last to struggling to accurately solve this problem of proper Prisma error => HttpError mapping in controllers on the server, but I just wonder if this is the PR to do it or we take a simpler "good enough" approach now like making the catch all 500, knowing we need to circle back? 🤷🏻 

That may be where I am leaning, agreeing with my comment above 😅  of changing base case to 500, and adding a `TODO` comment and new Issue to better handle Prisma errors to HttpErrors.

Reply:
So this is the direction I was thinking of taking: https://github.com/wasp-lang/wasp/pull/376/commits/08372fdbec58644bff09b7fa58bde2420071d9f2 Extract out the checking logic into `utils.js` so we can reuse in other controllers, but not do the full job just yet to keep PR more narrow. Handle the duplicate user email and `PrismaClientValidationError` but punt on rest with `500` as suggested. Thoughts on this for now? Thanks!

Reply:
I think this sounds great -> I agree that solving mapping of Prisma errors to Http codes is a too big task for this PR and we should do it later. I also agree it makes sense to do it at some point.

I am guessing Prisma pplz haven't done it because this mapping might somewhat depend on the use case -> it is to some level up to us to decide what stuff is internal server error and what is ok to pass to the user. On the other hand, I am guessing most use cases are 90% similar regarding how the mapping should go.

I also wonder in our case if such central mapping will be sufficient, or if we might be needing to have local mappings for specific parts of the code. But, I think this is good direction for now and we can see from here!

Commit looks good! I would probably immediatelly separate it into utils/prisma.js though, or maybe even into `core` -> why core? Because I am not sure if this is a general logic, that could be part of external library, in which case it could be part of utils.js, or if it is specific for our app, in which case it could go under `core`. But again, I am probably rushing too much -> utils sounds good for now and we can see how it develops.

TLDR; -> Sounds good, I agree with everything!